groclient/lib.py

```diff
@@ -912,6 +913,122 @@ def reverse_geocode_points(access_token: str, api_host: str, points: list):
     return r.json()["data"]
 
 
+def validate_series_object(series_object):
+    required_entities = {"item_id", "metric_id", "frequency_id", "source_id"}
+
+    for entity in required_entities:
+        if entity not in series_object or not series_object[entity]:
+            raise ValueError(f"{entity} is required and supposed to be a positive integer.")
+
+    invalid_atts = set(series_object.keys()).difference(required_entities)
+    if len(invalid_atts):
+        raise ValueError(f"Unsupported fields: {invalid_atts}.")
+
+
+def generate_payload_for_v2_area_weighting(
+    series: Dict[str, int],
+    region_id: int,
+    weights: Optional[List[Dict[str, int]]] = None,
+    weight_names: Optional[List[str]] = None,
+    start_date: Optional[str] = None,
+    end_date: Optional[str] = None,
+    method: Optional[str] = "sum",
+):
+    payload = {
+        "region_id": region_id,
+        "method": method,
+    }
+
+    # validate series and weights selection
+    try:
+        validate_series_object(series)
+    except ValueError as error:
+        raise ValueError(f"Failed to parse series selection: {error}")
+    payload["series"] = series
+
+    if weights and len(weights):
+        if weight_names:
+            raise ValueError(f"weights and weight_names are mutually exclusive. Please specify only one.")
+        try:
+            for weight in weights:
+                validate_series_object(weight)
+        except ValueError as error:
+            raise ValueError(f"Failed to parse weight selections: {error}")
+        payload["weights"] = weights
+    else:
+        if not weight_names or not len(weight_names):
+            raise ValueError(f"Please specify either weights or weight_names in params.")
+        payload["weight_names"] = weight_names
+
+    # add optional attrs
+    if start_date:
+        payload["start_date"] = start_date
+    if end_date:
+        payload["end_date"] = end_date
+
+    return json.dumps(payload)
+
+
+def format_v2_area_weighting_response(response_content: Dict[str, Any]) -> pd.DataFrame:
+    try:
+        data_points = response_content["data_points"]
+        weighted_series_df = pd.DataFrame(data_points)
+
+        # convert unix timestamps and rename date cols
+        datetime_col_mappings = {
+            "start_date": "timestamp", # add start_date col which is equivalent to end_date
```

Comment:
added because of the sample snippet. double check on whether we should keep it or not, for it's a duplicate of 'end_date'
@muzigao-gro @sahuguet

Reply:
Yes. All other end points return both start and end date. This is for consistency.